examples/integrations/oci/atp-cdi/src/main/java/io/helidon/examples/integrations/oci/atp/cdi/AtpResource.java

```diff
@@ -94,5 +124,61 @@ public Response generateWallet() {
 
         return Response.status(Response.Status.OK).entity(returnEntity).build();
     }
+
+    /**
+     * Returns SSLContext based on cwallet.sso in wallet.
+     *
+     * @param walletContent
+     * @return SSLContext
+     */
+    private static SSLContext getSSLContext(byte[] walletContent) throws IllegalStateException {
+        SSLContext sslContext = null;
+        try (ZipInputStream zis = new ZipInputStream(new BufferedInputStream(new ByteArrayInputStream(walletContent)))) {
+            ZipEntry entry = null;
+            while ((entry = zis.getNextEntry()) != null) {
+                if (entry.getName().equals("cwallet.sso")) {
+                    KeyStore keyStore = KeyStore.getInstance("SSO", new OraclePKIProvider());
+                    keyStore.load(zis, null);
+                    TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("PKIX");
+                    KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("PKIX");
+                    trustManagerFactory.init(keyStore);
+                    keyManagerFactory.init(keyStore, null);
+                    sslContext = SSLContext.getInstance("TLS");
+                    sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+                }
+                zis.closeEntry();
```

Comment:
Nit: does this need to be in a `finally` block?  Or does closing the input stream also close all open entries?

Reply:
Right: my point was, if something goes wrong in line 147, for example, yes, the stream _itself_ will be closed (by the try-with-resources block) but the _entry_ will not be closed (explicitly, anyway).  It's not clear to me whether this actually matters, of course, because perhaps closing the stream closes all entries so it's all moot.  On the other hand, if it does *not* matter, then maybe this `closeEntry` call should just be eliminated.  Or leave it as-is.